src/main/java/org/springframework/data/aerospike/cache/AerospikeCache.java

```
@@ -1,114 +1,144 @@
+						  AerospikeConverter aerospikeConverter,
+						  String namespace,
+						  String set,
+						  int expiration) {
 		this.client = client;
+		this.aerospikeConverter = aerospikeConverter;
 		this.namespace = namespace;
 		this.set = set;
-		this.createOnly = new WritePolicy(client.writePolicyDefault);
+		this.createOnly = new WritePolicy(client.getWritePolicyDefault());
 		this.createOnly.recordExistsAction = RecordExistsAction.CREATE_ONLY;
+		this.createOnly.expiration = expiration;
+		this.writePolicyForPut = new WritePolicy(client.getWritePolicyDefault());
+		this.writePolicyForPut.expiration = expiration;
 	}
 
-	protected Key getKey(Object key){
+	private Key getKey(Object key){
 		return new Key(namespace, set, key.toString());
 	}
 
-	private ValueWrapper toWrapper(Record record) {
-		return (record != null ? new SimpleValueWrapper(record.getValue(VALUE)) : null);
-	}
-
 	@Override
 	public void clear() {
-		// TODO Auto-generated method stub
+		client.truncate(null, namespace, set, null);
 	}
 
 	@Override
 	public void evict(Object key) {
-		this.client.delete(null, getKey(key));
+		client.delete(null, getKey(key));
+	}
 
+	@Override
+	public String getName() {
+		return AerospikeCacheUtils.getFullAerospikeCacheName(namespace, set);
 	}
 
 	@Override
-	public ValueWrapper get(Object key) {
-		Record record =  client.get(null, getKey(key));
-		ValueWrapper vr = toWrapper(record);
-		return vr;
+	public Object getNativeCache() {
+		return client;
 	}
 
-	@SuppressWarnings("unchecked")
 	@Override
-	public <T> T get(Object key, Class<T> type) {
-		return (T) client.get(null, getKey(key));
+	@SuppressWarnings("unchecked")
+	public <T> T get(Object key, Callable<T> valueLoader) {
+		T value = (T) client.get(null, getKey(key)).getValue(VALUE);
+		if (Objects.isNull(value)) {
+			try {
+				value = valueLoader.call();
+				if (Objects.nonNull(value)) {
+					put(key, value);
+				}
+			} catch (Exception e) {
+				log.warn("valueLoader exception", e);
```

according to the documentation from the interface:
```
If the valueLoader throws an exception, it is wrapped in a Cache.ValueRetrievalException
```
we should not swallow an exception and instead should wrap it into `Cache.ValueRetrievalException` and rethrow

---

also let's replace `catch (Exception e) {` with `catch (Throwable e) {`, as catching Exception type does not catch all exceptions